openapi/covid19/client.bal

```diff
@@ -24,13 +24,42 @@ public isolated client class Client {
     # Gets invoked to initialize the `connector`.
     # The connector initialization doesn't require setting the API credentials.
     #
-    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + config - The configurations to be used when initializing the `connector` 
     # + serviceUrl - URL of the target service 
     # + return - An error if connector initialization failed 
-    public isolated function init(http:ClientConfiguration clientConfig =  {}, string serviceUrl = "https://disease.sh") returns error? {
-        http:Client httpEp = check new (serviceUrl, clientConfig);
-        self.clientEp = httpEp;
-        return;
+    public isolated function init(ConnectionConfig config =  {}, string serviceUrl = "https://disease.sh") returns error? {
+
+        http:ClientConfiguration httpClientConfig = {
+           httpVersion: config.httpVersion,
+           timeout: config.timeout,
+           forwarded: config.forwarded,
+           poolConfig: config.poolConfig,
+           compression: config.compression,
+           circuitBreaker: config.circuitBreaker,
+           retryConfig: config.retryConfig
+        };
+        do {
+            if config.http1Settings is ClientHttp1Settings {
+                ClientHttp1Settings settings = check config.http1Settings.ensureType(ClientHttp1Settings);
+                httpClientConfig.http1Settings = {...settings};
+            }
+            if config.http2Settings is http:ClientHttp2Settings {
+                httpClientConfig.http2Settings = check config.http2Settings.ensureType(http:ClientHttp2Settings);
+            }
+            if config.cache is http:CacheConfig {
+                httpClientConfig.cache = check config.cache.ensureType(http:CacheConfig);
+            }
+            if config.responseLimits is http:ResponseLimitConfigs {
+                httpClientConfig.responseLimits = check config.responseLimits.ensureType(http:ResponseLimitConfigs);
+            }
+            if config.secureSocket is http:ClientSecureSocket {
+                httpClientConfig.secureSocket = check config.secureSocket.ensureType(http:ClientSecureSocket);
+            }
+            if config.proxy is http:ProxyConfig {
+                httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+            }
+        }
+        self.clientEp = check new (serviceUrl, httpClientConfig);
```

Comment:
```suggestion
        http:Client httpEp = check new (serviceUrl, httpClientConfig);
        self.clientEp = httpEp;
        return;
```

In other scenarios we have not directly assign values to `self.clientEp`. Shall we select one approach to maintain the consistency.

Reply:
Updated the PR